### PR TITLE
Task5

### PR DESCRIPTION
**Pull request by igorano**
task5

### CODE REVIEW

File: src/com/estafet/learning/US005/Task5.java [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+                System.out.format("Enter a number to view Fibonacci series: %n");
+                reader.next();
+            }
+            number = reader.nextInt();
+        } while (number <= 0);
+
+        int n1 = 0, n2 = 1, n3, i;
+        System.out.print(n1 + " " + n2);
+
+        for (i = 2; i < number; ++i) {
```
Comment (azankovskiy) [inline src/com/estafet/learning/US005/Task5.java:45]: what if `number == 1` ?

File: src/com/estafet/learning/US005/Task5.java [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+        long i, fact = 1;
+        for (i = 1; i <= number; i++) {
+            fact = fact * i;
+        }
+        System.out.format("Factorial of " + number + " is: " + fact + "%n");
+    }
+
+    public static void factorialOfOddNumbers() {
+        Scanner reader = new Scanner(System.in);
+        System.out.format("Please enter a positive and odd number to view Factorial of it: %n");
```
Comment (azankovskiy) [inline src/com/estafet/learning/US005/Task5.java]: I guess you should not filter even number on input. Instead, your method should calculate factorial, by omitting even numbers in sequence.
> to calculate the factorial of n, but to skip all even numbers

File: src/com/estafet/learning/US005/Task5.java (hunk 1 above)
Comment (igorano) [reply src/com/estafet/learning/US005/Task5.java:45]: image

File: src/com/estafet/learning/US005/Task5.java [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+        System.out.format("Input a year: ");
+        int year = input.nextInt();
+        switch (month) {
+            case 1:
+            case 3:
+            case 5:
+            case 12:
+            case 10:
+            case 8:
+            case 7:
```
Comment (azankovskiy) [inline src/com/estafet/learning/US005/Task5.java:140]: don't really like the idea from [the Oracle example](https://docs.oracle.com/javase/tutorial/displayCode.html?code=https://docs.oracle.com/javase/tutorial/java/nutsandbolts/examples/SwitchDemo2.java). Consider someone adds some assignment for a `case:` above one, that what you expected would assign a value, e.g. if I set `case 5: number_Of_DaysInMonth = 10;` what will be the value if `month = 3`

File: src/com/estafet/learning/US005/Task5.java (hunk 1 above)
Comment (azankovskiy) [reply src/com/estafet/learning/US005/Task5.java:45]: sorry, missed that in this case you print numbers before the loop statement.

File: src/com/estafet/learning/US005/Task5.java [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+        do {
+            while (!reader.hasNextInt()) {
+                System.out.format("That's not a number!%n");
+                System.out.format("Enter a positive number to view Factorial of it: %n");
+                reader.next();
+            }
+            number = reader.nextInt();
+        } while (number <= 0);
+
+        long i, fact = 1;
```
Comment (azankovskiy) [inline src/com/estafet/learning/US005/Task5.java]: variable `i` is not needed in the local scope, can be declared in loop `for (int i = 1; i <= number; i++)`

File: src/com/estafet/learning/US005/Task5.java [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+            System.out.format("Please enter a positive number!%n");
+            while (!reader.hasNextInt()) {
+                System.out.format("That's not a number!%n");
+                System.out.format("Enter a number to view Fibonacci series: %n");
+                reader.next();
+            }
+            number = reader.nextInt();
+        } while (number <= 0);
+
+        int n1 = 0, n2 = 1, n3, i;
```
Comment (azankovskiy) [inline src/com/estafet/learning/US005/Task5.java:42]: variable `i` is not needed in the local scope, can be declared in `for` loop below

File: src/com/estafet/learning/US005/Task5.java (hunk 4 above)
Comment (igorano) [reply src/com/estafet/learning/US005/Task5.java]: ok, changed.

File: src/com/estafet/learning/US005/Task5.java (hunk 5 above)
Comment (igorano) [reply src/com/estafet/learning/US005/Task5.java:42]: ok, changed.

**Comment by azankovskiy** [comment]
Done. Can be merged.

File: src/com/estafet/learning/US005/Task5.java (hunk 3 above)
Comment (igorano) [reply src/com/estafet/learning/US005/Task5.java:140]: changed logic

File: src/com/estafet/learning/US005/Task5.java (hunk 2 above)
Comment (igorano) [reply src/com/estafet/learning/US005/Task5.java]: changed.

File: src/com/estafet/learning/US005/Task5.java [hunk 6]
```diff
@@ -73,17 +73,20 @@ public static void factorial() {
 
     public static void factorialOfOddNumbers() {
         Scanner reader = new Scanner(System.in);
-        System.out.format("Please enter a positive and odd number to view Factorial of it: %n");
+        System.out.format("Please enter a positive number to view Factorial without even numbers in the sequence: %n");
         number = reader.nextInt();
 
-        if (number > 0 && number % 2 != 0) {
+        if (number > 0) {
             long i, fact = 1;
```
Comment (azankovskiy) [inline src/com/estafet/learning/US005/Task5.java]: why you declare local (method) vars (`long i`) for loops?

Comment (igorano) [reply src/com/estafet/learning/US005/Task5.java]: changed.

File: src/com/estafet/learning/US005/Task5.java [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+    public static void main(String[] args) {
+        fibonacii();
+        factorial();
+        checkNumberIsEvenOrOdd();
+        addTwoNumbers();
+        numberOfDaysInMonth();
+        factorialOfOddNumbers();
+    }
+
+    public static int number;
```
Comment (azankovskiy) [inline src/com/estafet/learning/US005/Task5.java]: can be local var

Comment (igorano) [reply src/com/estafet/learning/US005/Task5.java]: it is used by some of the methods

File: src/com/estafet/learning/US005/Task5.java [hunk 8]
```diff
@@ -12,7 +12,7 @@ public static void main(String[] args) {
         factorialOfOddNumbers();
     }
 
-    public static int number;
+    private static int number;
```
Comment (azankovskiy) [inline src/com/estafet/learning/US005/Task5.java]: I mean it's a member of the class. Do you need to share its value among methods? If no - declare it as a local (method) variable.

Comment (igorano) [reply src/com/estafet/learning/US005/Task5.java]: ok, changed.

**Pull request merged**